nix/docker.nix

```diff
@@ -0,0 +1,123 @@
+############################################################################
+# Docker image builder
+#
+# To test it out, use:
+#
+#   docker load -i $(nix-build -A dockerImage.base --no-out-link)
+#   docker load -i $(nix-build -A dockerImage.latency-tests --no-out-link)
+#   docker load -i $(nix-build -A dockerImage.mainnet --no-out-link)
+#   docker load -i $(nix-build -A dockerImage.mainnet-ci --no-out-link)
+#   docker load -i $(nix-build -A dockerImage.shelley_staging --no-out-link)
+#   docker load -i $(nix-build -A dockerImage.shelley_staging_short --no-out-link)
+#   docker load -i $(nix-build -A dockerImage.staging --no-out-link)
+#   docker load -i $(nix-build -A dockerImage.testnet --no-out-link)
+#
+# For all output except base, run:
+#
+#   docker run inputoutput/cardano-node:<TAG>
+#
+# ie:
+#
+#   docker run inputoutput/cardano-node:6a5b233e5861aa0012287a5557b3a6a08afbf3ca-testnet
+#
+# For base, you will need to provide all variables. ie:
+#
+#   docker run -ti \
+#   --volume $PATH_TO/state-docker:/data \
+#   --volume $PATH_TO/configuration-mainnet.yaml:/config/config.yaml \
+#   --volume $PATH_TO/mainnet-topology.json:/config/topology.json \
+#   --volume $PATH_TO/mainnet-genesis.json:/config/genesis.json \
+#   inputoutput/cardano-node:<GIT_HASH>-custom
+#   --genesis-hash <GENESIS_HASH>
+#
+############################################################################
+
+{ system ? builtins.currentSystem
+, crossSystem ? null
+, config ? {}
+, profiling ? false
+, pkgs ? import ../nix { inherit system crossSystem config profiling; }
+, dockerTools
+
+# The main contents of the image.
+, cardano-node
+, scripts
+
+# Get the current commit
+, gitrev ? pkgs.iohkNix.commitIdFromGitRepoOrZero ../.git
+
+# Other things to include in the image.
+, bashInteractive
+, cacert
+, coreutils
+, curl
+, glibcLocales
+, iana-etc
+, iproute
+, iputils
+, socat
+, utillinux
+
+, repoName ? "inputoutput/cardano-node"
+}:
+
+with pkgs;
```

Comment:
it's a bit strange to both require explicit packages arguments using `with pkgs;` which kind-of negate the point of explicit dependencies.
Can we remove `pkgs` from arguments?